### ISSUE TITLE
[Prometheus]: Field limit is too low

### ISSUE THREAD

**Issue by flash1293** [elastic/integrations#11466]
### Integration Name

Prometheus [prometheus]

### Dataset Name

prometheus.collector

### Integration Version

2.10.0

### Agent Version

-

### Agent Output Type

elasticsearch

### Elasticsearch Version

8.15

### OS Version and Architecture

-

### What did you do?

The prometheus integration is used to collect arbitrary metrics. Because of this, it's easily possible for the number of fields in the index to grow a lot.

The current default is to index the first 1000 fields, then ignore everything else and not map it.

This limit should be raised to 10000 to make sure larger amounts of metrics are searchable properly

### What did you see?

-

### What did you expect to see?

-

**Comment by lalit-satapathy** [comment]
Hi @flash1293, 

Thanks for the issue. Definitely Prometheus package, packages using prometheus input is a potential candidate. We can also check other packages with usage of dynamic mappings.

Also, considering that _index.mapping.total_fields.ignore_dynamic_beyond_limit_ is already set to _true_, is the impact somewhat less?

**Comment by flash1293** [comment]
> Also, considering that index.mapping.total_fields.ignore_dynamic_beyond_limit is already set to true, is the impact somewhat less?

@lalit-satapathy yes, the docs won't be dropped, but random metrics can end up being ignored which means they are essentially lost to the user.
